### PR TITLE
feat(ci)!: remove Debian 9 and Amazon Linux 1 testing MONGOSH-1542

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: packages/build/src/barque.ts [hunk 1]
```diff
@@ -311,10 +309,8 @@ export class Barque {
         return `${base}/apt/ubuntu/dists/focal/mongodb-${edition}/${packageFolderVersion}/multiverse/binary-${targetArchitecture}/${packageFileName}`;
       case 'ubuntu2204':
         return `${base}/apt/ubuntu/dists/jammy/mongodb-${edition}/${packageFolderVersion}/multiverse/binary-${targetArchitecture}/${packageFileName}`;
-      case 'debian92':
-        return `${base}/apt/debian/dists/buster/mongodb-${edition}/${packageFolderVersion}/main/binary-${targetArchitecture}/${packageFileName}`;
       case 'debian10':
-        return `${base}/apt/debian/dists/stretch/mongodb-${edition}/${packageFolderVersion}/main/binary-${targetArchitecture}/${packageFileName}`;
+        return `${base}/apt/debian/dists/buster/mongodb-${edition}/${packageFolderVersion}/main/binary-${targetArchitecture}/${packageFileName}`;
```
Comment (addaleax) [inline packages/build/src/barque.ts:313]: This mixup seems to have been around since basically forever :)

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64de0cd2850e6178dd746200/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64de35ee0ae606a187ea9de1/tasks

**Review by mcasimir — APPROVED** [review]

**Comment by mcasimir** [comment]
nice cleanup

**Pull request merged**